[v1.5][F14-B4] Bibliotek

takuyakanbr opened the pull request:
@qiu-siqi @592363789 @fishTT

takuyakanbr committed:
Add tests for storage of Book and BookShelf

takuyakanbr committed:
Add reading and saving book shelf to Storage and StorageManager

takuyakanbr committed:
Fix unused imports in storage tests

takuyakanbr committed:
Fix typo in storage test data

qiu-siqi committed:
Merge pull request #49 from takuyakanbr/update-storage

Add storage for Book and BookShelf

qiu-siqi committed:
Update UndoableCommand to support books

qiu-siqi committed:
Update CommandTestUtil and CliSyntax

qiu-siqi committed:
Update UndoableCommandTest

qiu-siqi committed:
Update UndoCommandTest

qiu-siqi committed:
Update RedoCommandTest

qiu-siqi committed:
Update DeleteCommand

qiu-siqi committed:
Remove unused import statements

qiu-siqi committed:
Update DeleteCommandTest

qiu-siqi committed:
Rename method in Command

qiu-siqi committed:
Change text in ExitCommand

qiu-siqi committed:
Update ClearCommand

qiu-siqi committed:
Update ClearCommandTest

qiu-siqi committed:
Change AddCommand

qiu-siqi committed:
Change AddCommandParser

qiu-siqi committed:
Add BookShelfParser

qiu-siqi committed:
Change tests and parsers for add command

qiu-siqi committed:
Change constant names in TypicalIndexes

qiu-siqi committed:
Update ListCommand

qiu-siqi committed:
Update ListCommandTest

qiu-siqi committed:
Update SelectCommand

qiu-siqi committed:
Update SelectCommandTest

qiu-siqi committed:
Rename constants in commons/core/Messages

qiu-siqi committed:
Ignore EditCommandTests temporarily

qiu-siqi committed:
Fix style issues

fishTT committed:
Deleted yl_coder.jpg

qiu-siqi committed:
Merge pull request #52 from fishTT/Jianghao

Deleted yl_coder.jpg

qiu-siqi committed:
Remove unrequired info

takuyakanbr committed:
Merge pull request #53 from qiu-siqi/fix-dg-issue

* Remove unrequired info in DG Glossary
* Fixes 50

takuyakanbr committed:
Change style used for documentation

takuyakanbr committed:
Move UG and DG table of content to left sidebar

takuyakanbr committed:
Merge branch 'master' into change-doc-style

qiu-siqi committed:
Fix style issues

qiu-siqi committed:
Change fxml files for UI

qiu-siqi committed:
Add UI classes to support books

qiu-siqi committed:
Update guihandles to support books

qiu-siqi committed:
Update GuiTestAssert

qiu-siqi committed:
Update UI tests

qiu-siqi committed:
Update BrowserPanel

qiu-siqi committed:
Update MainWindow in UI

qiu-siqi committed:
Update MainApp

qiu-siqi committed:
Update TestApp

qiu-siqi committed:
Rename and update AddressBookSystemTest to BibliotekSystemTest

takuyakanbr committed:
Adjust asciidoc styling

* Use anchors for section headers
* Change size of User Stories table in DG
* Adjustments to stylesheet used for asciidoc

takuyakanbr committed:
Add newline in asciidoctor.css

takuyakanbr committed:
Adjust asciidoc styling of table of content

* Remove redundant font imports in gh-pages.css

takuyakanbr committed:
Fix font import in asciidoctor.css to use https

qiu-siqi committed:
Remove EditCommand and FindCommand

qiu-siqi committed:
Fix style issues

takuyakanbr committed:
Merge pull request #55 from takuyakanbr/change-asciidoc-style

Change asciidoc styling

takuyakanbr committed:
Merge pull request #56 from qiu-siqi/update-commands

Update commands to support books

qiu-siqi committed:
Merge branch 'morph' into update-ui

qiu-siqi committed:
Fix all unit tests

qiu-siqi committed:
Change formatting of BookListCard

qiu-siqi committed:
Fix style issues

qiu-siqi committed:
Fix style issues

qiu-siqi committed:
Remove guihandles and UI tests for person

qiu-siqi committed:
Fix unit tests

takuyakanbr committed:
Add network component

* Add class GoogleBooksApi that allows searching books and obtaining getting book details using Google Books API
* Add request and result events for searching books using Google Books API
* Add new dependency async-http-client

takuyakanbr committed:
Add json deserializers

* Add methods and deserializers to convert JSON response from Google Books API into a Book / BookShelf

takuyakanbr committed:
Remove getAuthorSet and getCategorySet from SampleDataUtil

takuyakanbr committed:
Add tests for json deserializers

takuyakanbr committed:
Add tests for GoogleBooksApi

* Add wrapper classes for AsyncHttpClient and Response from async-http-client to allow easier unit testing

takuyakanbr committed:
Add request and result events for obtaining book details

qiu-siqi committed:
Fix comments and style

takuyakanbr committed:
Merge pull request #57 from qiu-siqi/update-ui

Update UI component to support books

takuyakanbr committed:
Fix issues in network component

* Remove 'Google' from the name of API request and result events
* Use String.format to create request urls from url constants

takuyakanbr committed:
Fix content type check in GoogleBooksApi

qiu-siqi committed:
Merge pull request #58 from takuyakanbr/add-google-books-api

Add Network component

takuyakanbr committed:
Fix system tests

takuyakanbr committed:
Fix more tests

takuyakanbr committed:
Remove deprecated parts of the code

takuyakanbr committed:
Fix errors in ParserUtilTest

takuyakanbr committed:
Remove unused command prefixes / test strings

takuyakanbr committed:
Remove unused fxml files

takuyakanbr committed:
Fix issues in AddCommandSystemTest and DeleteCommandSystemTest

qiu-siqi committed:
Merge pull request #59 from takuyakanbr/fix-tests

Fix tests and remove deprecated code

takuyakanbr committed:
Add additional tests for the model component

takuyakanbr committed:
Fix BookShelfTest to use static imports for typical books

qiu-siqi committed:
Merge pull request #60 from takuyakanbr/add-model-tests

Add additional tests for the model component

takuyakanbr committed:
Add search command

takuyakanbr committed:
Update MainApp to start and stop the network component

takuyakanbr committed:
Add UI support for displaying search results

takuyakanbr committed:
Add model support for displaying search results

takuyakanbr committed:
Add logic support for displaying search results

* Add, list, redo, undo commands will always switch to displaying the user's book list
* Clear and delete commands will fail unless the user's book list is displayed
* Select command will select the corresponding item in the currently displayed list

takuyakanbr committed:
Add unit and integration tests for search command

takuyakanbr committed:
Add unit tests for clear, delete, select commands

* Add unit tests for clear, delete, select commands to verify behavior when displayed list is search results list

liuyang1213 commented on src/main/java/seedu/address/commons/events/network/ApiBookDetailsRequestEvent.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        return "book details for: " + bookId;
```
Such "magical" string should be avoided :)

liuyang1213 reviewed (commented):
Excellent!

takuyakanbr committed:
Add system tests for search command

takuyakanbr committed:
Improve search and select command system tests

592363789 committed:
for v1.1 change the photo name to githubname.jpg

592363789 committed:
change jpg to png

qiu-siqi committed:
Change photo name and format

qiu-siqi committed:
Change photo name and format

qiu-siqi committed:
Merge pull request #63 from 592363789/v1.1phote

for v1.1 change the photo name to githubname.jpg

takuyakanbr committed:
Merge pull request #64 from qiu-siqi/photo

Change photo name and format

takuyakanbr committed:
Update user guide to add search command

qiu-siqi committed:
Update diagrams in DG

takuyakanbr committed:
Merge pull request #67 from qiu-siqi/update-diagrams

Update diagrams in DG

takuyakanbr committed:
Implement request changes

takuyakanbr committed:
Fix import order for MainWindow

qiu-siqi committed:
Change introduction of UG

takuyakanbr committed:
Use a different way to switch between book list and search results

qiu-siqi committed:
Merge pull request #62 from takuyakanbr/add-search-command

Add search command

qiu-siqi committed:
Update commands in UG

qiu-siqi committed:
Resolve merge conflicts

takuyakanbr committed:
Add additional tests for network component

takuyakanbr committed:
Update readme

* Update UI image
* Remove learning outcomes

takuyakanbr committed:
Add tests for NetworkManager

qiu-siqi committed:
Add more components to Book, Update Deserializer, Change AddCommand

qiu-siqi committed:
Update classes with dependency on Book, Update tests for AddCommand

qiu-siqi committed:
Remove unused variable

takuyakanbr committed:
Merge pull request #69 from takuyakanbr/add-network-tests

Add additional tests for network component

takuyakanbr committed:
Merge pull request #70 from takuyakanbr/update-readme

Update readme

qiu-siqi committed:
Make suggested changes

qiu-siqi committed:
Make style changes

qiu-siqi committed:
Add tests for XmlAdaptedBook

qiu-siqi committed:
Change BookDeserializer to reject Book when no ISBN

qiu-siqi committed:
Add InvalidBookException, Fix style

takuyakanbr committed:
Merge pull request #72 from qiu-siqi/add-command

Change add command and add more components to Book

takuyakanbr committed:
Add book details panel

* Add book details panel
* Update CSS to support book details panel
* Replace browser panel with book details panel in the UI

qiu-siqi committed:
Add more commands to UG

takuyakanbr committed:
Add unit and integration test for book details panel

takuyakanbr committed:
Remove browser panel and related test code

takuyakanbr committed:
Add a light-blue based theme

* Use the light theme by default

takuyakanbr committed:
Move selection change event handlers into MainWindow

takuyakanbr committed:
Merge pull request #73 from takuyakanbr/update-ui

Add book details panel

takuyakanbr committed:
Merge pull request #74 from takuyakanbr/add-light-theme

Add a light-blue based theme

qiu-siqi committed:
Add more commands into UG

takuyakanbr committed:
Merge pull request #68 from qiu-siqi/update-UG

Update User Guide

qiu-siqi committed:
Remove RedoCommand

takuyakanbr committed:
Remove fishTT.png

takuyakanbr committed:
Add fishtt.png

* Update image url in AboutUs

takuyakanbr committed:
Merge pull request #75 from qiu-siqi/delete-redo

Remove RedoCommand

takuyakanbr committed:
Merge pull request #76 from takuyakanbr/fix-aboutus

Fix AboutUs

takuyakanbr committed:
Merge branch 'master' into dev

takuyakanbr committed:
Bump library versions

takuyakanbr committed:
Fix setting of default width and height

takuyakanbr committed:
Bump version to v1.1

takuyakanbr committed:
Fix null exception in GuiSettings

takuyakanbr committed:
Merge pull request #77 from takuyakanbr/release-v1.1

Release v1.1

takuyakanbr committed:
Update DeveloperGuide to add feature contribution

takuyakanbr committed:
Merge pull request #81 from takuyakanbr/update-dg

Update DeveloperGuide to add feature contribution

qiu-siqi committed:
Add contribution to developer guide

takuyakanbr committed:
Merge pull request #82 from qiu-siqi/update-DG

Add contribution to developer guide

592363789 committed:
Add contribution to DeveloperGuide (#83)

takuyakanbr committed:
Make book details panel scroll to top when selected book changes (#79)

takuyakanbr committed:
Add theme command and update UI styling (#80)

* Add theme command
* Add unit, integration, and system tests for theme command
* Add new 'white' theme
* Adjust UI styling

592363789 committed:
Add rating field to Book (#66)

fishTT committed:
Add contribution to DeveloperGuide (#84)

takuyakanbr committed:
Refactor network component (#88)

* Refactor network component
* Add new test dependency: mockito
* Use mocks for testing NetworkManager and GoogleBooksApi

qiu-siqi committed:
Add recent command (#86)

* Add recent command
* Store recently selected books
* Add recentbooks.xml into gitignore
* Update User Guide

takuyakanbr committed:
Improve system tests (#85)

* Move the duplicated assertCommandFailure methods in the system test classes into BibliotekSystemTest
* Assert that add command does not change the selected search results card
* Minor changes to comments

qiu-siqi committed:
Add recent command to implementation in DG (#92)

takuyakanbr committed:
Update DG to describe implementation of search feature (#93)

takuyakanbr committed:
Bump version to v1.2 (#94)

takuyakanbr committed:
Update build scripts (#95)

* Upgrade gradle version from 2.12 to 4.6
* Update shadow, coveralls, testfx versions
* Add configuration for CircleCI
* Skip install step for Travis builds
* Remove unused DummySearchPage

takuyakanbr committed:
Run junit tests in parallel (#96)

* Run junit tests in parallel, depending on number of processors
* Randomize file name of temp files to avoid interference during parallel testing

takuyakanbr committed:
Add Priority and Status fields to Book  (#97)

* Add Priority and Status fields to Book
* Update relevant tests
* Update UI to display status, priority, and rating

qiu-siqi committed:
Update User Guide and Developer Guide (#102)

takuyakanbr committed:
Update badges and library list in README

qiu-siqi committed:
Merge pull request #105 from takuyakanbr/update-readme

Update badges and library list in README

qiu-siqi committed:
Fix selection issues due to recent (#104)

Fixes #103

qiu-siqi committed:
Handle xxxSelectionChangedEvent in LogicManager (#106)

Fixes #101

takuyakanbr committed:
Update list command to allow filtering and sorting  (#100)

* Update model to allow for sorting the display book list
* Update list command to allow filtering and sorting
* Add tests for list command and list command parser

qiu-siqi committed:
Make ClearCommand clear BookDetailsPanel (#107)

Fixes #99

liuyang1213 commented:
It's great to use UML diagrams in the developer guide we have learned so far. One thing I would like to mention is that try to apply standard UML notations. For instance, in [this diagram](https://github.com/CS2103JAN2018-F14-B4/main/blob/master/docs/images/SearchCommandResultSequenceDiagram.png), "returns control arrow" of method calls is missing. Although return value is omittable sometimes, return control arrow should always be there. Overall, very nice work! Carry on!

qiu-siqi committed:
Make BookCard when in view (#108)

* Improve book list performance.

Ported from upstream: [se-edu/addressbook-level4](https://github.com/se-edu/addressbook-level4/commit/a4ef5229e29bb1da91d462a650ee82c243ab1b7c).

qiu-siqi committed:
Add command to view book reviews (#112)

- Add ReviewsCommand
- Update relevant tests
- Update user guide

Fixes #24

takuyakanbr committed:
Reduce time taken to compare UI book lists against book array (#114)

takuyakanbr committed:
Refactor network component (#113)

* Refactor network component
* Instead of using request and result events, network API methods now return CompletableFuture

qiu-siqi committed:
Make some changes in tests (#115)

takuyakanbr committed:
Fix tests (#118)

* Fix add and search command tests
* Increase memory limit for gradle test processes

takuyakanbr committed:
Update dev ops section of developer guide (#117)

* Add instructions for setting up CircleCI
* Make external links open in a new window

takuyakanbr committed:
Bump version to v1.3 (#119)

takuyakanbr committed:
Use randomized file name for recent books file when testing (#121)

592363789 committed:
Add edit command (#111)

* Add edit command to edit the status, priority, and rating of books
* Add unit and integration tests for edit command

Fixes #18 
Fixes #20 
Fixes #25

takuyakanbr committed:
Fix appveyor test issues (#123)

* Edit GuiTestAssert to navigate to card for each index
* Update appveyor config

qiu-siqi committed:
Display book description in WebView (#116)

Fixes #87

takuyakanbr committed:
Edit reviews command (#120)

* Process the page using javascript after loading
* Clear the page using javascript before loading reviews for another book
* Don't make BookReviewsPanel post NewResultAvailableEvents

takuyakanbr committed:
Bump TestFX version (#126)

* Update testfx versions

* Remove unused testfx-legacy

* Fix height adjustment in BookDescriptionView

takuyakanbr committed:
Disable the command box when a long-running action is active  (#127)

Fixes #110

takuyakanbr committed:
Add book image to book details panel (#129)

takuyakanbr committed:
Add icons to author and category tags (#130)

* Limit number of category tags in list to 4 per book

takuyakanbr committed:
Keep the ordering of authors and categories as per original API data (#131)

* Use lists instead of sets to store authors and categories.
* Update Book constructor to accept Collection instead of Set of authors and categories.

takuyakanbr committed:
Annotate code for collation and add output of collate (#133)

qiu-siqi committed:
Add collate (#136)

* Add collate

* Add collate files

takuyakanbr committed:
Update README and user guide (#137)

takuyakanbr committed:
Bump version to v1.4 (#139)

takuyakanbr committed:
Add command aliasing system (#150)

* Update model to support aliases and add tests

* Update storage to support loading and saving of alias list and add tests

* Update UI to display a list of aliases and add tests

* Add addalias, deletealias, and aliases commands and add tests

* Add system tests for addalias, deletealias, and aliases commands

* Add addalias, deletealias, and aliases commands to user guide

fishTT committed:
Add welcome panel (#132)

* Add a welcome panel that displays the app logo, name, and a randomized quote

takuyakanbr committed:
Add additional tests (#153)

qiu-siqi committed:
Add library command (#138)

* Add library command
* Add tests for library command

Fixes #26

takuyakanbr committed:
Update user guide (#154)

* Update user guide

qiu-siqi committed:
Update Developer Guide (#155)

* Add introduction
* Update diagrams
* Do some renaming and remove irrelevant information

qiu-siqi committed:
Add getActiveList in model, update and refactor commands (#156)

592363789 committed:
Collate code (#151)

takuyakanbr committed:
Improve UI consistency across OSes (#160)

takuyakanbr committed:
Fix saving of window position when UI is minimized (#161)

takuyakanbr committed:
Restrict the number of help windows to 1 (#162)

takuyakanbr committed:
Update search command (#163)

* Rename 'search term' to 'key words'
* Request for 40 books from API and limit to 30 when processing them

qiu-siqi committed:
Fix use of WebViewManager (#164)

takuyakanbr committed:
Adjust UI styling and sizing (#165)

qiu-siqi committed:
Remove duplicated code in XmlRecentBooksStorage, update class diagram (#167)

qiu-siqi committed:
Use only one class for left panel (#169)

* Remove `SearchResultsPanel` and `RecentBooksPanel`
* Update events
* Update tests

qiu-siqi committed:
Make commands handle undo logic themselves (#170)

* Add `undoLogic` in `UndoableCommand`s
* Make undo message reflect undone command
* Update DG
Fixes #149

qiu-siqi committed:
Add command auto correction (#171)

Limitations: 
* Correcting of command word only
* Will not correct for extra space in command word
* Distance of 1 character away; will not correct for swapping of characters

fishTT committed:
Collate code (#176)

qiu-siqi committed:
Add PPP (#175)

* Make partial changes to DG

592363789 committed:
Update documentation (#157)

fishTT committed:
PPP (#177)

592363789 committed:
Make commands case-insensitive (#179)

takuyakanbr committed:
Add project portfolio page (#180)

* Updated developer guide
* Updated asciidoctor stylesheet

qiu-siqi committed:
Add Events to fix command issues (#178)

takuyakanbr committed:
Bump version to v1.5rc (#181)

takuyakanbr committed:
Add delay to makeGetRequest on event of exception (#183)

qiu-siqi committed:
Store recentbooks in UserPrefs rather than Config (#185)

* Change default path of recentbooks.xml

qiu-siqi committed:
Change behaviour for pressing Enter in CommandBox (#184)

* Accept autocorrection using Enter
* If no auto corrected command, Enter shows nothing.

qiu-siqi committed:
Add more tests for add and recent commands (#186)

qiu-siqi committed:
Update Developer Guide (#187)

* Add prefaces
* Update diagrams
* Add implementation of `library` command

qiu-siqi committed:
Update User Guide (#188)

* Add prefaces
* Make new sections
* Move things around

fishTT committed:
Add hint feature (#159)

Fixes #33

takuyakanbr committed:
Improve various parts of the code and tests  (#190)

fishTT committed:
Add test for case-insensitive commands (#192)

takuyakanbr committed:
Add tests for hint system (#194)

fishTT committed:
Update collated code (#197)

takuyakanbr committed:
Update collated code (#195)

takuyakanbr committed:
Fix consistency issue with filtering and sorting (#198)

* Ensure that applying the same filters and sort mode will not cause the ordering of the books to change.
Fixes #189

592363789 committed:
Add encryption/decryption of book shelf data (#134)

Fixes #41
Fixes #128

qiu-siqi committed:
Refactor some stuff (#199)

* Mainly put command autocorrection into a seperate class.

takuyakanbr committed:
Improve security of encryption system (#200)

qiu-siqi committed:
Make minor changes to Developer Guide (#204)

qiu-siqi committed:
Add some tests  (#201)

qiu-siqi committed:
Update User Guide (#203)

takuyakanbr committed:
Add command alias system to developer guide (#207)

592363789 committed:
Add some tests and collate for test (#206)

qiu-siqi committed:
Finalize collate (#208)

qiu-siqi committed:
Finalize PPP (#205)

takuyakanbr committed:
Add more tests for lock system (#210)

takuyakanbr committed:
Fix various issues (#209)

qiu-siqi committed:
Refactor a method and make style changes (#211)

* Minor refactoring
* Update collate

592363789 committed:
update ugdg and PPP (#193)

takuyakanbr committed:
Update all collated code (#212)

takuyakanbr committed:
Update sample book shelf (#214)

qiu-siqi committed:
Add testing instructions (#215)

takuyakanbr committed:
Bump version to v1.5 (#213)

takuyakanbr committed:
Fix various bugs (#216)

takuyakanbr committed:
Bump version to v1.5.1 (#217)
